fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/identifiers/FedoraId.java

```diff
@@ -342,6 +342,14 @@ public FedoraId asDescription() {
         return FedoraId.create(getBaseId(), appendHashIfPresent(FCR_METADATA));
     }
 
+    /**
+     * Returns the FullId of the described resource.
+     * @return The ID.
+     */
+    public String getFullDescribedId() {
```

Comment:
You call this method a fair bit. For performance reasons, maybe the "full described id" should be stored in a field either in `FedoraId` or `SparqlTranslateVisitor`?